fix(linter): report correct diagnostic count on minified files

camc314 opened the pull request:
fixes #13865

testing this is tricky as we have a different impl for cfg test which means this code path is never hit. but i tested this in vscode repo and appears to work as expected

camc314 commented on crates/oxc_diagnostics/src/service.rs:
```diff
@@ -195,7 +196,8 @@ impl DiagnosticService {
                                 .unwrap();
                         }
-                        break;
+                        is_minified = true;
+                        continue;
```
it's outside the `for diagnostic of diagnostics` loop which are diagnostics we get from processing one single file?

overlookmotel commented:
I'm not sure if this is correct or not.

I can confirm that the number of warnings + errors reported in summary is now the same with / without `--silent`.

But... the number in summary does not correspond to the actual number of warnings/errors output.

In Kibana repo, after this PR:

```
% node path/to/oxc/apps/oxlint/dist/cli.js
<itemized 16881 warnings, 8 errors>
Found 64036 warnings and 8 errors.
Finished in 2.9s on 76140 files with 88 rules using 14 threads.

% node path/to/oxc/apps/oxlint/dist/cli.js --silent
Found 64036 warnings and 8 errors.
Finished in 1.9s on 76140 files with 88 rules using 14 threads.
```

Previously:

```
% node path/to/oxc/apps/oxlint/dist/cli.js
<itemized 16881 warnings, 8 errors>
Found 16881 warnings and 6 errors.
Finished in 2.5s on 76138 files with 88 rules using 14 threads.

% node path/to/oxc/apps/oxlint/dist/cli.js --silent
Found 64036 warnings and 6 errors.
Finished in 3.0s on 76138 files with 88 rules using 14 threads.
```

Is this intentional?

(I'm unclear why we skip linting minified files anyway, I'd have expected user should exclude any files they don't want linted)

overlookmotel commented:
### Merge activity

* **Sep 21, 3:14 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camc314 committed:
fix(linter): report correct diagnostic count on minified files (#13896)

fixes #13865

testing this is tricky as we have a different impl for cfg test which means this code path is never hit. but i tested this in vscode repo and appears to work as expected
